### PR TITLE
Add cibuildwheel deployment

### LINKED ISSUE #20: Add cibuildwheels deployment
Once we have the package out on PyPi, we should add in relevant automated wheel deployments.

Here we'll need to add cibuildwheels stuff.

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #20 

Changes made in this Pull Request:

 - 
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by IAlibay** [comment]
This _should_ do the trick, it'll need testing in an actual deployment though.

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Build sdist
+        run: pipx run build --sdist
+
+      - name: upload artifacts
```
Comment (ianmkenney) [inline .github/workflows/deploy.yaml:71]: From this section, do we get the wheels in PRs where tags are pushed?

Comment (IAlibay) [reply .github/workflows/deploy.yaml:71]: What do you mean by "in PRs"?

When you push tags the wheels will be uploaded to testpypi directly so you can pull them down and test as necessary. You'll also get the artifacts stored locally but that's the exact same thing as whatt's on testpypi (unless the testpypi upload failed).

**Comment by IAlibay** [comment]
To better explain how this work:

1. When you want to do a release you tie off all the necessary changelog etc.. 
2. You git push some tags based on the final state of the main branch
3. As soon as you push the tags, this CI action is triggered and it'll build all the wheels and upload them to testpypi
4. The action will then pull down the wheels from testpypi and test them
5. If everything returns green then you can go ahead and do a full release
6. Publish a github release and this CI action gets triggered again
7. This time it'll create all the wheels and upload them to PyPi
8. From there your release is done and you're good to go.

**Comment by ianmkenney** [comment]
Alright sounds good, I'm good to merge whenever then!

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (ianmkenney) [reply .github/workflows/deploy.yaml:68]: Oh, oops, that's why you're removing the pull trigger.

**Pull request merged**